### PR TITLE
Policy split

### PR DESCRIPTION
**Pull request by spinal84**
Modify policy files according to API docs

### CODE REVIEW

File: policy/policy_always_online.c [hunk 1]
```diff
@@ -440,47 +441,47 @@ icd_policy_init(struct icd_policy_api *policy_api,
...
 
   data->always_change = gconf_client_get_bool(gconf,
                                               CHANGE_WHILE_CONNECTED_KEY, NULL);
 
   ILOG_INFO("always online connection change: %s",
             data->always_change ? "Yes" : "No");
 
-  data->search_interval = gconf_client_get_int(gconf, SEARCH_INTERVAL_KEY,
+  data->timeout = gconf_client_get_int(gconf, SEARCH_INTERVAL_KEY,
                                                NULL);
```
Comment (freemangordon) [inline policy/policy_always_online.c]: fix indentation (in a follow-up patch)

File: policy/policy_always_online.c [hunk 2]
```diff
@@ -69,11 +69,8 @@ get_iap_count()
 }
 
 static void
-cancel_always_online_timer(gpointer user_data)
+policy_always_online_cancel_timer(struct always_online_data *data)
 {
-  struct always_online_data *data =
```
Comment (freemangordon) [inline policy/policy_always_online.c]: data not used?

File: policy/policy_always_online.c [hunk 3]
```diff
@@ -123,11 +120,12 @@ parse_device_mode_ind(DBusMessage *message,
...
       {
         ILOG_INFO("always online: normal mode");
-        cancel_always_online_timer(data);
+        policy_always_online_cancel_timer(data);
         data->timeout_id =
-            g_timeout_add(2000, always_online_timer_cb, data);
+            g_timeout_add(2000, policy_always_online_make_request_cb, data);
 
-        ILOG_INFO("always online waiting 2s for the normal mode to propagate through the rest of the system, timer id is %d",
+        ILOG_INFO("always online waiting 2s for the normal mode to propagate "
```
Comment (freemangordon) [inline policy/policy_always_online.c]: strings may cross 80 chars boundary to ease grepping in the source code (just a note)

File: policy/policy_always_online.c [hunk 4]
```diff
@@ -418,22 +422,23 @@ icd_policy_always_online_connected(struct icd_policy_request *network,
   if (data->always_change && data->priority)
   {
     if (data->priority(network->service_type, network->service_id,
-                          network->network_type, network->network_attrs,
-                          &highest_network_priority))
+                       network->network_type, network->network_attrs,
+                       &highest_network_priority) )
     {
-      ILOG_DEBUG("always online timer not cancelled because higher priority network exists and always_change is set (%s/0x%04x/%s/%d)",
+      ILOG_DEBUG("always online timer not cancelled because higher priority "
```
Comment (freemangordon) [inline policy/policy_always_online.c]: ditto

File: policy/policy_always_online.c [hunk 5]
```diff
@@ -493,76 +573,19 @@ icd_policy_init(struct icd_policy_api *policy_api,
...
-  if (!message)
+  if (!data->flightmode_signals ||
+      !policy_always_online_flightmode_init(data) ||
+      !policy_always_online_gconf_init(data) )
   {
-    ILOG_CRIT("always online could not create flightmode request message");
-    goto failed;
+    ILOG_CRIT("always online failed to connect, always online disabled");
+    policy_always_online_destruct((gpointer *)&data);
+        return;
```
Comment (freemangordon) [inline policy/policy_always_online.c]: indentation

File: policy/policy_change.c [hunk 6]
```diff
@@ -145,8 +151,8 @@ icd_policy_change_new_request(struct icd_policy_request *new_request,
 
     if (new_request->attrs & ICD_POLICY_ATTRIBUTE_BACKGROUND)
     {
-      ILOG_INFO("policy change not accepted for req %p with ICD_POLICY_ATTRIBUTE_BACKGROUND",
-                new_request);
+      ILOG_INFO("policy change not accepted for req %p "
```
Comment (freemangordon) [inline policy/policy_change.c]: fixme

File: policy/policy_change.c [hunk 7]
```diff
@@ -174,14 +180,15 @@ icd_policy_change_new_request(struct icd_policy_request *new_request,
...
-      policy_done(ICD_POLICY_ACCEPTED, data);
+      policy_change_do_cb(ICD_POLICY_ACCEPTED, data);
     }
     else
     {
       if (policy_attrs & ICD_POLICY_ATTRIBUTE_NO_INTERACTION)
       {
-        ILOG_INFO("policy change cannot ask for dialog since ICD_POLICY_ATTRIBUTE_NO_INTERACTION set");
-        policy_done(ICD_POLICY_REJECTED, data);
+        ILOG_INFO("policy change cannot ask for dialog "
```
Comment (freemangordon) [inline policy/policy_change.c]: fixme

File: policy/policy_merge.c [hunk 8]
```diff
@@ -18,19 +18,20 @@ string_equal(const char *a, const char *b)
...
+  icd_policy_request_merge_fn merge_requests = *private;
   const GSList *l;
   struct icd_policy_request *request;
 
   if (new_request->attrs & ICD_POLICY_ATTRIBUTE_ALWAYS_ONLINE_CHANGE &&
-      string_equal(new_request->network_id, OSSO_IAP_ANY))
+      policy_merge_string_equal(new_request->network_id, OSSO_IAP_ANY))
   {
-    ILOG_DEBUG("policy merge allows OSSO_IAP_ANY since changing while connected is allowed");
+    ILOG_DEBUG("policy merge allows OSSO_IAP_ANY since changing "
```
Comment (freemangordon) [inline policy/policy_merge.c]: fixme

File: policy/policy_merge.c [hunk 9]
```diff
@@ -70,8 +73,8 @@ icd_policy_merge_new_request(struct icd_policy_request *new_request,
   {
     if (new_request->attrs & ICD_POLICY_ATTRIBUTE_BACKGROUND)
     {
-      ILOG_INFO("policy merge got req %p with attribute ICD_POLICY_ATTRIBUTE_BACKGROUND, rejecting it",
-                new_request);
+      ILOG_INFO("policy merge got req %p with attribute "
```
Comment (freemangordon) [inline policy/policy_merge.c]: fixme

**Comment by freemangordon** [comment]
I am ok with the changes, nits will be fixed in a follow-up patch

**Comment by MerlijnWajer** [comment]
Looks good to me. :)

**Pull request merged**